### PR TITLE
issue-376: refactor who is using svelte

### LINKED ISSUE #376: Refactor 'Who is using Svelte'
Move data into JSON and then use an each block inside of svelte component.

I'll do the refactor :)

### PR DESCRIPTION
**Pull request by dreitzner**
resolves #376 

- [x]  change existing PRs

### CODE REVIEW

**Comment by swyxio** [comment]
Deploy Preview is done!

Built with commit 9f661eb993effd2181ed8f49284fbcf1e838952e

https://deploy-preview-377--svelte-community.netlify.app

**Comment by dreitzner** [comment]
may need to port existing PRs

**Comment by peopledrivemecrazy** [comment]
I like this change 👍

**Comment by swyxio** [comment]
big change. i dont mind it but dont know the downsides. @antony and @benmccann would this be ok for you guys pulling it in?

**Comment by antony** [comment]
I like it, but this part of the site is an ongoing discussion - and I'm not sure W what we agreed to do with it yet.

However this is an improvement on the existing so I'm game.

File: whos-using-svelte/README.md [hunk 1]
```diff
@@ -2,7 +2,26 @@
...
+
+### JSON format
+`href`, `src`, `alt` are required.
+
+`style` and `picture` are optional.
+```json
+{
+    "href":"https://your-company.com",
+    "src": "organisations/yourCompanyLogo.svg",
+	"alt":"Your Company logo",
```
Comment (benmccann) [inline whos-using-svelte/README.md]: it looks like some of these lines use tabs and some use spaces

File: whos-using-svelte/WhosUsingSvelte.js [hunk 2]
```diff
@@ -0,0 +1,830 @@
+export const companies = [
+		{
+			"href":"https://1password.com",
```
Comment (benmccann) [inline whos-using-svelte/WhosUsingSvelte.js]: I'm not sure why `src` has a space after it and the other attributes don't. I'd put a space after all of them for consistency

File: whos-using-svelte/WhosUsingSvelte.js [hunk 3]
```diff
@@ -0,0 +1,830 @@
...
+		},
+		{
+			"href":"https://gieson.com/Library/projects/utilities/timetrek/",
+			"style":"background-color:#4774b3;",
+			"src": "organisations/gieson.svg",
+			"alt":"Gieson TimeTrek Logo"
+		},
+		{
+			"href":"https://iwillcodeit.io",
+			"src": "https://iwillcodeit.io/images/logo-letters-only.svg",
```
Comment (benmccann) [inline whos-using-svelte/WhosUsingSvelte.js]: remove this one since it's external

File: whos-using-svelte/WhosUsingSvelte.js [hunk 4]
```diff
@@ -0,0 +1,830 @@
...
+			"alt":"Tproger («Типичный программист»)" 
+		},
+		{
+			"href":"https://transloadit.com",
+			"src": "organisations/transloadit.svg",
+			"alt":"Transloadit" 
+		},
+		{
+			"href":"https://seagage.com",
+			"src": "https://static.wixstatic.com/media/3e1d33_2e8f3c427913446ea9091cdc903cd4fb~mv2.png/v1/fill/w_192,h_38,al_c,q_85,usm_0.66_1.00_0.01/3e1d33_2e8f3c427913446ea9091cdc903cd4fb~mv2.webp",
```
Comment (benmccann) [inline whos-using-svelte/WhosUsingSvelte.js]: remove this one since it's external

File: whos-using-svelte/WhosUsingSvelte.js [hunk 5]
```diff
@@ -0,0 +1,830 @@
...
+			"alt":"Creative Tim" 
+		},
+		{
+			"href":"https://turgensec.com",
+			"src": "organisations/TurgenSec.png",
+			"alt":"TurgenSec" 
+		},
+		{
+			"href":"https://yatagarasu.vercel.app/",
+			"src": "https://yt3.ggpht.com/a/AATXAJy5ifhPq1BR87Hhh3ltx5swo5tn_uylAmsL5ZAQMg=s900-c-k-c0xffffffff-no-rj-mo",
```
Comment (benmccann) [inline whos-using-svelte/WhosUsingSvelte.js]: remove this one since it's external

File: whos-using-svelte/WhosUsingSvelte.js [hunk 6]
```diff
@@ -0,0 +1,830 @@
...
+			"alt":"Yatagarasu logo"
+		},
+		{
+			"href":"https://oliveoilseason.com",
+			"src": "https://static.oliveoilseason.com/g/images/logo/2020/191219_OOS_Logo__120w.png",
```
Comment (benmccann) [inline whos-using-svelte/WhosUsingSvelte.js]: remove this one since it's external

**Comment by benmccann** [comment]
I don't have any objections to this. I do think we'll remove 90% of these logos and just keep the prominent ones or do something else to clean up this section. But I guess there's no harm in making this change in the meantime

**Comment by peopledrivemecrazy** [comment]
> I don't have any objections to this. I do think we'll remove 90% of these logos and just keep the prominent ones or do something else to clean up this section. But I guess there's no harm in making this change in the meantime

Randomize logos that are displayed and a dedicated link to list all of them on another page?

File: whos-using-svelte/WhosUsingSvelte.js (hunk 2 above)
Comment (dreitzner) [reply whos-using-svelte/WhosUsingSvelte.js]: Fixed. Must have been a misthap in my search and replace

File: whos-using-svelte/README.md (hunk 1 above)
Comment (dreitzner) [reply whos-using-svelte/README.md]: #done

File: whos-using-svelte/WhosUsingSvelte.js (hunk 3 above)
Comment (dreitzner) [reply whos-using-svelte/WhosUsingSvelte.js]: #done

File: whos-using-svelte/WhosUsingSvelte.js (hunk 4 above)
Comment (dreitzner) [reply whos-using-svelte/WhosUsingSvelte.js]: #done

File: whos-using-svelte/WhosUsingSvelte.js (hunk 5 above)
Comment (dreitzner) [reply whos-using-svelte/WhosUsingSvelte.js]: #done

File: whos-using-svelte/WhosUsingSvelte.js (hunk 6 above)
Comment (dreitzner) [reply whos-using-svelte/WhosUsingSvelte.js]: #done

**Comment by dreitzner** [comment]
> I don't have any objections to this. I do think we'll remove 90% of these logos and just keep the prominent ones or do something else to clean up this section. But I guess there's no harm in making this change in the meantime

We could later on add a property prominent to the object and display only those and then have a "see all" button

**Comment by dreitzner** [comment]
I added sorting, I hope this is welcome. If not, it is easily removed again :)

**Comment by dreitzner** [comment]
@sw-yx 

- all comments should be resolved
- merged upstream (removal of external images)
- implemented sorting by `alt`

**Comment by swyxio** [comment]
thank you so much @dreitzner :)

**Comment by peopledrivemecrazy** [comment]
Good job! 🚀 @dreitzner

**Comment by dreitzner** [comment]
Follow up issue: https://github.com/sveltejs/svelte/issues/5917

PR: https://github.com/sveltejs/svelte/pull/5918

**Comment by benmccann** [comment]
Thanks for this!     

>Randomize logos that are displayed and a dedicated link to list all of them on another page?
>We could later on add a property prominent to the object and display only those and then have a "see all" button

Maybe show the "prominent" logos and then have a carousel beneath it that's scrolling through the others?

**Comment by swyxio** [comment]
we're having the "marquee" user or "prominent" logo discussion over here https://github.com/sveltejs/community/issues/135#issuecomment-765797447

**Pull request merged**